From 8dd2ea9c3e9e6166cb3d52d502c5942663997037 Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 17 Dec 2014 21:39:10 -0500
Subject: [drm] nouveau/bios: memset dcb struct to zero before parsing

Message-id: <1418852380-13061-60-git-send-email-rclark@redhat.com>
Patchwork-id: 102209
O-Subject: [RHEL7 drm 59/89] drm/nouveau/bios: memset dcb struct to zero before parsing
Bugzilla: 1173317
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1173317
Upstream: 3.17

commit 1edd504622feb46de07065c212ed01136470bd67

Author:     Ben Skeggs <bskeggs@redhat.com>
AuthorDate: Mon Sep 8 10:33:32 2014 +1000
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Wed Dec 10 17:45:35 2014 -0500

    drm/nouveau/bios: memset dcb struct to zero before parsing

    commit 595d373f1e9c9ce0fc946457fdb488e8a58972cd upstream

    Fixes type/mask calculation being based on uninitialised data for VGA
    outputs.

    Signed-off-by: Ben Skeggs <bskeggs@redhat.com>

Signed-off-by: Rob Clark <rclark@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/nouveau/core/subdev/bios/dcb.c b/drivers/gpu/drm/nouveau/core/subdev/bios/dcb.c
index 88606bf..bd8d348 100644
--- a/drivers/gpu/drm/nouveau/core/subdev/bios/dcb.c
+++ b/drivers/gpu/drm/nouveau/core/subdev/bios/dcb.c
@@ -124,6 +124,7 @@ dcb_outp_parse(struct nouveau_bios *bios, u8 idx, u8 *ver, u8 *len,
         struct dcb_output *outp)
 {
  u16 dcb = dcb_outp(bios, idx, ver, len);
+ memset(outp, 0x00, sizeof(*outp));
  if (dcb) {
   if (*ver >= 0x20) {
    u32 conn = nv_ro32(bios, dcb + 0x00);
-- 
1.7.1